### PR TITLE
Add more toOpenArray

### PR DESCRIPTION
**Pull request by data-man**

### CODE REVIEW

File: lib/system.nim [hunk 1]
```diff
@@ -4414,6 +4414,18 @@ when not defined(js):
...
+  template toOpenArray*[T](x: seq[T]): openarray[T] =
+    toOpenArray(x, 0, x.high)
+  template toOpenArray*[T](x: openarray[T]): openarray[T] =
+    toOpenArray(x, 0, x.high)
+  template toOpenArray*[I, T](x: array[I, T]): openarray[T] =
+    toOpenArray(x, 0, x.high)
+  template toOpenArrayByte*(x: string): openarray[byte] =
+    toOpenArrayByte(x, 0, x.high)
+  template toOpenArrayChar*(x: string): openarray[char] =
+    toOpenArrayChar(x, 0, x.high)
```
Comment (krux02) [inline lib/system.nim]: Why do we need all these templates? Aren't they the default automatic conversion rules?

Comment (data-man) [reply lib/system.nim]: Thanks, toOpenArrayChar removed.

File: lib/system.nim [hunk 2]
```diff
@@ -4414,6 +4414,14 @@ when not defined(js):
     magic: "Slice".}
   proc toOpenArrayByte*(x: string; first, last: int): openarray[byte] {.
     magic: "Slice".}
+  template toOpenArray*[T](x: seq[T]): openarray[T] =
```
Comment (hlaaftana) [inline lib/system.nim]: So if you overrode toOpenArray(openarray) with a new template, this would get called for seq instead of the new override? Why is this here? Same for array override below

Also wouldn't `x.high` evaluate `x` for the second time? That would be confusing. Not to judge but bad PR idea in my opinion
